src/sagemaker_training/smdataparallel.py

```diff
@@ -143,7 +143,7 @@ def _get_mpirun_command(
             "-x",
             smdataparallel_flag,
             "-x",
-            "FI_PROVIDER=sockets",
+            "FI_PROVIDER=efa",
```

Comment:
Should we have a flag here to toggle the FI_PROVIDER?

Reply:
- If the machine is equipped with EFA hardware and FI_PROVIDER=efa is set, then it will use EFA for communication
- If the machine is not equipped with EFA hardware and FI_PROVIDER=efa is set, then it will fallback to sockets by defaults for communication.

Reply:
True, but how do I force to use "sockets" if the machine is equipped with EFA hardware?

Reply:
No, you can't. If the user spins up the instance that is capable of using EFA hardware then it will always use EFA. It is by design.